src/Screen/AsSource.php

```diff
@@ -18,8 +18,9 @@ trait AsSource
      */
     public function getContent(string $field)
     {
-        return Arr::get($this->toArray(), $field)
-            ?? Arr::get($this->getRelations(), $field)
-            ?? $this->$field;
+        if ($this->isRelation($field) && !$this->relationLoaded($field)) {
```

Comment:
These lines will break the usual behavior that users are accustomed to. For example with access to private properties to be prohibited, it was enough to change only
 the call to `$this->$field` to` $this->getAttribute($field) `

Reply:
Ok, do you mean when passing $field === null?

in that case just prepend:

if($field === null){
return $this->toArray();
}

Reply:
I mean, for solution #1989 it is enough to change the call to `getAttribute`.
Other changes do not affect issue #1989. Plus, they break current tests (and existing user code.)

https://github.com/orchidsoftware/platform/blob/39d631aa2c09c247bfe45d802ff2ff81448ef15b/tests/Unit/Screen/SourceTest.php#L67-L91

Reply:
Oh, I see, it must support dot notation.